manifests/networkd.pp

```diff
@@ -5,10 +5,53 @@
 # @param ensure The state that the ``networkd`` service should be in
 # @param path path where all networkd files are placed in
 # @param manage_all_network_files if enabled, all networkd files that aren't managed by puppet will be purged
+# @param link_profiles
+#   Hash of network link profiles that can be referenced by it's key on an interface
+#   The structure is equal to the 'link' parameter of an interface.
+# @param netdev_profiles
+#   Hash of netdev profiles that can be referenced by it's key on an interface
+#   The structure is equal to the 'netdev' parameter of an interface.
+# @param network_profiles
+#   Hash of network profiles that can be referenced by it's key on an interface
+#   The structure is equal to the 'network' parameter of an interface.
+# @param interfaces
+#  Hash of interfaces to configure on a node.
+#  The link, netdev and network parameters are deep merged with the respective profile
+#  (referenced by the key of the interface).
+#  With the profiles you can set the default values for a network.
+#  Hint: to remove a profile setting for an interface you can either overwrite or
+#  set it to '~' for removal.
+#  Example (hiera yaml notation):
```

Comment:
I think these should be listed as `@example`.

Also these new `systemd::networkd` docs are missing from `REFERENCE.md` for some reason.

Reply:
the `@example` seems to be usable only once per class/define since it's an example for the parameter and not the whole class I'm not changing. (and it's still in private api ! )